src/base/cub_ddl_log.c

```diff
@@ -0,0 +1,1054 @@
+	    }
+	}
+
+      len = cub_create_log_mgs (buf);
+      if (len < 0)
+	{
+	  goto write_error;
+	}
+
+      if (fwrite (buf, 1, len, fp) != len)
+	{
+	  goto write_error;
+	}
+
+      if (strcmp (ddl_audit_handle->app_name, "loaddb") == 0)
+	{
+	  cub_file_copy (ddl_audit_handle->file_name, dest_path);
+	}
+
+      if ((UINT64) ftell (fp) > prm_get_bigint_value (PRM_ID_DDL_AUDIT_LOG_SIZE))
+	{
+	  fclose (fp);
+	  cub_ddl_log_backup (ddl_audit_handle->log_filepath);
+	  fp = NULL;
+	}
+    }
+
+write_error:
+  if (fp != NULL)
+    {
+      fclose (fp);
+      fp = NULL;
+    }
+}
+
+static int
+cub_create_log_mgs (char *msg)
+{
+  int retval = 0;
+  char result[20] = { 0 };
+
+  if (strcmp (ddl_audit_handle->app_name, "loaddb") == 0)
+    {
+      if (ddl_audit_handle->err_code < 0)
+	{
+	  snprintf (result, sizeof (result), "ERROR:%d", ddl_audit_handle->err_code);
+
+	}
+      else
+	{
+	  strcpy (result, "OK");
+	}
+
+      retval = snprintf (msg, DDL_LOG_BUFFER_SIZE, "%s %d|%s|%s|%s\n",
+			 ddl_audit_handle->execute_start_time,
+			 ddl_audit_handle->pid, ddl_audit_handle->user_name, result, ddl_audit_handle->schema_file);
+    }
+  else if (strcmp (ddl_audit_handle->app_name, "csql") == 0)
+    {
+      if (ddl_audit_handle->err_code < 0)
+	{
+	  snprintf (result, sizeof (result), "ERROR:%d", ddl_audit_handle->err_code);
+	  ddl_audit_handle->elapsed_time[0] = '\0';
+	}
+      else
+	{
+	  strcpy (result, "OK");
+	}
+
+      retval = snprintf (msg, DDL_LOG_BUFFER_SIZE, "%s %d|%s|%s|%s|%s\n",
+			 ddl_audit_handle->execute_start_time,
+			 ddl_audit_handle->pid,
+			 ddl_audit_handle->user_name,
+			 result, ddl_audit_handle->elapsed_time, ddl_audit_handle->sql_text);
+    }
+  else
+    {
+      if (ddl_audit_handle->err_code < 0)
+	{
+	  snprintf (result, sizeof (result), "ERROR:%d", ddl_audit_handle->err_code);
+	}
+      else
+	{
+	  strcpy (result, "OK");
+	}
+
+
+      if (ddl_audit_handle->log_type & DDL_LOG_NO_ELAPSED_TIME)
+	{
+	  snprintf (ddl_audit_handle->elapsed_time, sizeof (ddl_audit_handle->elapsed_time), "elapsed time 0.000");
+	}
+
+      retval = snprintf (msg, DDL_LOG_BUFFER_SIZE, "%s %s|%s|%s|%s|%s|%s\n",
+			 ddl_audit_handle->execute_start_time,
+			 ddl_audit_handle->ip_addr,
+			 ddl_audit_handle->user_name,
+			 result, ddl_audit_handle->elapsed_time, ddl_audit_handle->msg, ddl_audit_handle->sql_text);
+
+    }
+
+  if (retval >= DDL_LOG_BUFFER_SIZE)
+    {
+      msg[DDL_LOG_BUFFER_SIZE - 2] = '\n';
+      msg[DDL_LOG_BUFFER_SIZE - 1] = '\0';
+      retval = DDL_LOG_BUFFER_SIZE;
+    }
+  return retval;
+}
+
+static void
+cub_ddl_log_backup (const char *path)
+{
+  char backup_file[PATH_MAX] = { 0 };
+
+  snprintf (backup_file, sizeof (backup_file), "%s.bak", path);
+
+  unlink (backup_file);
+  rename (path, backup_file);
+}
+
+static FILE *
+cub_fopen_and_lock (const char *path, const char *mode)
+{
+#define MAX_RETRY_COUNT 100
+  int retry_count = 0;
+  FILE *result = NULL;
+  retry_count = 0;
```

Comment:
You already initialized it at definition (remove this line).

Reply:
Thank you, i changed the code which you commented.